enhancements/installer/vsphere-ipi-multiple-datacenter-clusters.md

```diff
@@ -0,0 +1,964 @@
+  // +kubebuilder:validation:MaxLength=80
+  VMGroupName string `json:"vmGroupName"`
+
+  // hostGroupName is the Host Gorup name configured
+  // within a vCenter cluster defining a group
+  // of ESXi hosts.
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=80
+  HostGroupName string `json:"hostGroupName"`
+}
+
+// VSpherePlatformVCenterSpec stores the vCenter connection fields
+// https://github.com/kubernetes/cloud-provider-vsphere/blob/master/pkg/common/config/types_yaml.go
+type VCenterSpec struct {
+
+  // server is the fully-qualified domain name or the IP address of the vCenter server.
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MaxLength=255
+  Server string `json:"server"`
+
+  // port is the TCP port that will be used to communicate to
+  // the vCenter endpoint. This is typically unchanged from
+  // the default of HTTPS TCP/443.
+  // +kubebuilder:validation:Optional
+  // +kubebuilder:validation:Minimum=1
+  // +kubebuilder:validation:Maximum=32767
+  // +kubebuilder:default=443
+  Port uint `json:"port,omitempty"`
+
+  // User is the username that will be used to connect to vCenter
+  // +kubebuilder:validation:Required
+  User string `json:"user"`
+
+  // Password is the password for the user to use to connect to the vCenter.
+  // +kubebuilder:validation:Required
+  Password string `json:"password"`
+
+  // Datacenter in which VMs are located.
+  // +kubebuilder:validation:Required
+  // +kubebuilder:validation:MinItems=1
+  Datacenters []string `json:"datacenters"`
+
+}
+
+// VSpherePlatformPlacementConstraint is the context information for VM placements within a failure domain
+type PlacementConstraint struct {
+  // resourcePool is the absolute path of the resource pool where virtual machines will be
+  // created. The absolute path is of the form /<datacenter>/host/<cluster>/Resources/<resourcepool>.
+  // +kubebuilder:validation:Optional
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=2048
+  ResourcePool string `json:"resourcePool,omitempty"`
+
+  // folder is the name or inventory path of the folder in which the
+  // virtual machine is created/located.
+  // +kubebuilder:validation:Optional
+  // +kubebuilder:validation:MinLength=1
+  // +kubebuilder:validation:MaxLength=2048
+  Folder string `json:"folder,omitempty"`
+}
+
+// +kubebuilder:validation:Enum="";Allowed;NotAllowed
+type DeploymentSuitable string
+
+const (
+    // Allowed indicates that the deployment is suitable for
+    // control plane nodes.
+    Allowed DeploymentSuitable = "Allowed"
+
+    // NotAllowed indicates that the deployment is not suitable for
+    // control plane nodes.
+    NotAllowed DeploymentSuitable = "NotAllowed"
+)
+
+// DeploymentZoneSpec holds the association between a
+// vCenter, failure domain and the virtual machine placementConstraints
+type DeploymentZoneSpec struct {
```

Comment:
Is this entire section duplicate of API section above with some subtle differences?

Reply:
Yes, the platform spec in the installer will need generally the same API as openshift/api config infra